CMakeLists.txt

```diff
@@ -0,0 +1,707 @@
+# Government or Lawrence Livermore National Security, LLC, and shall not
+# be used for advertising or product endorsement purposes.
+#
+##############################################################################
+
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+
+###
+# grab the version string
+###
+file(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/VERSION SILO_VERSION)
+
+###-----------------------------------------------------------------------------
+# project command will automatically create cmake vars for major, minor,
+# patch and tweak from passed VERSION
+###-----------------------------------------------------------------------------
+project(Silo VERSION ${SILO_VERSION} LANGUAGES CXX C)
+
+###-----------------------------------------------------------------------------
+# location for Silo CMake includes
+###-----------------------------------------------------------------------------
+set(CMAKE_MODULE_PATH ${Silo_SOURCE_DIR}/CMake)
+
+###-----------------------------------------------------------------------------
+# If not already set, use a default build type of Release
+###-----------------------------------------------------------------------------
+
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
+endif()
+
+###-----------------------------------------------------------------------------
+# Output directories.
+###-----------------------------------------------------------------------------
+
+if(NOT DEFINED CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY lib CACHE INTERNAL "Output directory for libraries.")
+endif()
+
+if(NOT DEFINED CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
+    set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY lib CACHE INTERNAL "Output directory for archives.")
+endif()
+
+if(NOT DEFINED CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY bin CACHE INTERNAL "Output directory for executables.")
+endif()
+
+# turn on folders (VS IDE, ignored otherwise)
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+#
+# create an include dir for generated include files
+#
+set(silo_build_include_dir ${Silo_BINARY_DIR}/include)
+file(MAKE_DIRECTORY ${silo_build_include_dir})
+
+###---------------------------------------------------------------------------
+# User options
+###---------------------------------------------------------------------------
+
+option(SILO_ENABLE_SHARED  "Build silo shared library." ON)
+option(SILO_ENABLE_SILEX "Enable building of silex (requires Qt5)" OFF)
+option(SILO_ENABLE_BROWSER "Enable building of browser" ON)
+option(SILO_ENABLE_FORTRAN "Enable Fortran interface to Silo" ON)
+option(SILO_ENABLE_HDF5 "Enable hdf5 support" ON)
+option(SILO_ENABLE_JSON "Enable experimental json features" OFF)
+option(SILO_ENABLE_PYTHON_MODULE "Enable python module" OFF)
+option(SILO_ENABLE_TESTS "Enable building of tests." OFF)
```

Comment:
Do I have to re-cmake if I want to run tests but neglected to set `SILO_ENABLE_TESTS` to true?

Reply:
Yes. 

However, I made the pascal config-site cmake file set it to true, so it would always be on when built there. 

If you would prefer it to default to true, I can change that.